src/H5Shyper.c

```diff
@@ -10297,7 +10297,7 @@ H5Sselect_hyperslab(hid_t space_id, H5S_seloper_t op, const hsize_t start[], con
  REVISION LOG
 --------------------------------------------------------------------------*/
 herr_t
-H5S_combine_hyperslab(H5S_t *old_space, H5S_seloper_t op, const hsize_t start[], const hsize_t *stride,
+H5S_combine_hyperslab(const H5S_t *old_space, H5S_seloper_t op, const hsize_t start[], const hsize_t *stride,
```

Comment:
It looks like the ref count reached by old_space is modified, so I don't think that this should be changed to be const.

Reply:
Where is the modification?  Wouldn't that have caused a compiler error after I made it `const`?

Reply:
Line 10397:
 old_space->select.sel_info.hslab->span_lst->count--;

I would have thought that the compiler would barf also, but possibly the multiple levels of indirection are at play.

Reply:
`span_lst` isn't `const`, which I believe explains it. Tagging `old_space` as const doesn't promise infinite recursion of constness. And code is already passing `const H5S_t`s to this function, hence the warnings. So I still think this change is right.

Reply:
Yeah, realized that later in the evening. :-)